lib/Doctrine/Common/Proxy/Autoloader.php

```diff
@@ -49,7 +49,11 @@ public static function resolveFile($proxyDir, $proxyNamespace, $className)
             throw InvalidArgumentException::notProxyClass($className, $proxyNamespace);
         }
 
-        $className = str_replace('\\', '', substr($className, strlen($proxyNamespace) + 1));
+        // remove proxy namespace from class name
+        $className = substr($className, strlen($proxyNamespace));
```

Comment:
Ah, yep, that is indeed true. Indeed irrelevant, as doctrine always puts a `\\` between the proxy namespace and the class name.

Thanks, indeed flawed example by me